### PR TITLE
[SYCLomatic][Doc] Add doc description for analysis mode

### PR DESCRIPTION
**Pull request by ziranzha**

### CODE REVIEW

**Review by zhimingwang36 — APPROVED** [review]
LGTM

File: docs/dev_guide/migration/analysis-mode.rst [hunk 1]
```diff
@@ -0,0 +1,43 @@
+Analysis Mode Guide
+=======================
+
+Analysis Mode generates a summary report that shows how many lines of code are migrated with confidence and how many lines of code are not migrated perfectly due to the language gap between CUDA and SYCL, also it provides estimation of human efforts required to complete the code migration based on the effort levels. Three levels are defined for effort: low, medium, and high. All migration issues are classified into them. Effort required for each level: Typically, low effort level means developer can solve the issues in minutes, medium effort level issues take hours and if high effort level issue is reported, unfortunately, it means there is no equivalent functionality provided yet in SYCL or SYCL ecosystem libraries for the CUDA API or the CUDA library API, developer need manual rewrite the functionality.
```
Comment (lapel22) [inline docs/dev_guide/migration/analysis-mode.rst]: Suggest change to "Analysis Mode generates a summary report showing how many lines of code are migrated with confidence, and how many lines of code are migrated imperfectly due to the language gap between CUDA and SYCL. The report also provides an estimation of how much human effort will be required to complete the code migration. Three "effort levels" have been defined: low effort, medium effort, and high effort. All migration issues are classified into these levels. Typically, a low effort level indicates issues that the developer can solve in minutes, whereas a medium effort level indicates an issue that may take hours to solve. If a high effort level issue is reports, this unfortunately means that there is not yet an equivalent functionality provided in SYCL or SYCL ecosystem libraries for the CUDA API or the CUDA library API.  In these cases, the developer will need to manually rewrite the functionality."

File: docs/dev_guide/migration/analysis-mode.rst [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+Following is an example of analysis report:
```
Comment (lapel22) [inline docs/dev_guide/migration/analysis-mode.rst]: Change to "An example analysis report can be seen as follows:"

**Review by lapel22 — CHANGES_REQUESTED** [review]
Reviewed and left comments!

File: docs/dev_guide/migration/analysis-mode.rst (hunk 2 above)
Comment (zhimingwang36) [reply docs/dev_guide/migration/analysis-mode.rst]: fixed.

**Pull request merged**